__tests__/app.test.js

```diff
@@ -47,38 +47,96 @@ describe('/api', () => {
+        test('GET:404 responds with an appropriate status code and error message when given a valid but non-existent id', () => {
+            return request(app)
+            .get('/api/articles/9999')
+            .expect(404)
+            .then(({ body: { msg } }) => {
+                expect(msg).toBe("Resource not found.");
+            });
+        });
+        test('GET:400 responds with an appropriate status code and error message when given an invalid id', () => {
+            return request(app)
+            .get('/api/articles/notAnId')
+            .expect(400)
+            .then(({ body: { msg } }) => {
+                expect(msg).toBe("Bad request.");
+            });
         });
     });
-    test('GET:400 responds with an appropriate status code and error message when given an invalid id', () => {
-        return request(app)
-        .get('/api/articles/notAnId')
-        .expect(400)
-        .then(({ body: { msg } }) => {
-            expect(msg).toBe("Bad request.");
+    describe('PATCH', () => {
+        test('PATCH:200 updates an existing article by parameterised article_id', () => {
+            const update = {
+                inc_votes: 2
+            };
+    
+            return request(app)
+            .patch('/api/articles/5')
+            .expect(200)
+            .send(update)
+            .then(({ body: { article } }) => {
+                expect(article).toEqual({
+                    article_id: 5,
+                    title: "UNCOVERED: catspiracy to bring down democracy",
+                    topic: "cats",
+                    author: "rogersop",
+                    body: "Bastet walks amongst us, and the cats are taking arms!",
+                    votes: 2,
+                    created_at: '2020-08-03T13:14:00.000Z',
+                    article_img_url:
+                        "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700",
+                });
+            });
+        });
+        test('PATCH:400 responds with an appropriate status code and error message when provided with a malformed body', () => {
+            return request(app)
+            .patch('/api/articles/5')
+            .expect(400)
+            .send({})
+            .then(({ body: { msg } }) => {
+                expect(msg).toBe("Bad request.");
+            });
+        });
+        test('PATCH:400 responds with an appropriate status code and error message when attempting to update database with an incorrect type ', () => {
+            return request(app)
+            .patch('/api/articles/5')
+            .expect(400)
+            .send({
+                inc_votes: 'hello'
+            })
+            .then(({ body: { msg } }) => {
+                expect(msg).toBe("Bad request.");
+            });
+        });
+        test('PATCH:400 responds with an appropriate status code and error message when attempting to update a non-existent article', () => {
```

Comment:
What if someone tries to vote on an an article with an id that is the incorrect type e.g. /api/articles/hello ?

Reply:
This one should be a 404 - it's not that the request is invalid, it's that the id cannot be found in our data.